File: src/main/java/dev/mantasboro/trakt5/entities/BaseShow.java
package dev.mantasboro.trakt5.entities;

import org.threeten.bp.OffsetDateTime;

import java.util.List;

public class BaseShow {

    public Show show;

    /** collection, watched */
    public List<BaseSeason> seasons;

    /** collection */
    public OffsetDateTime last_collected_at;
    /** watchlist */
    public OffsetDateTime listed_at;
    /** watched */
    public Integer plays;
    public OffsetDateTime last_watched_at;
    public OffsetDateTime last_updated_at;
    public OffsetDateTime reset_at;
    /** progress */
    public Integer aired;
    public Integer completed;
    public List<Season> hidden_seasons;
    public Episode next_episode;
    public Episode last_episode;

}

File: src/main/java/dev/mantasboro/trakt5/entities/BaseShow.kt
package dev.mantasboro.trakt5.entities

import org.threeten.bp.OffsetDateTime

data class BaseShow(
    val show: Show? = null,

    /** collection, watched  */
    val seasons: List<BaseSeason>? = null,

    /** collection  */
    val last_collected_at: OffsetDateTime? = null,

    /** watchlist  */
    val listed_at: OffsetDateTime? = null,

    /** watched  */
    val plays: Int? = null,
    val last_watched_at: OffsetDateTime? = null,
    val last_updated_at: OffsetDateTime? = null,
    val reset_at: OffsetDateTime? = null,

    /** progress  */
    val aired: Int? = null,
    val completed: Int? = null,
    val hidden_seasons: List<Season>? = null,
    val next_episode: Episode? = null,
    val last_episode: Episode? = null,
)